pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -1850,10 +1850,9 @@ public static MessageMetadata peekMessageMetadata(ByteBuf metadataAndPayload, St
             long consumerId) {
         try {
             // save the reader index and restore after parsing
-            int readerIdx = metadataAndPayload.readerIndex();
+            metadataAndPayload.markReaderIndex();
             MessageMetadata metadata = Commands.parseMessageMetadata(metadataAndPayload);
-            metadataAndPayload.readerIndex(readerIdx);
-
+            metadataAndPayload.resetReaderIndex();
```

Comment:
How this change be relevant?

Reply:
+1, I think we can keep this PR only focused on the revert job. If this is a part need to improve, it's better to create a separate PR.